core/src/main/java/hudson/ExtensionFinder.java

```diff
@@ -550,6 +558,29 @@ public Object get() {
             public List<IndexItem<?, Object>> getLoadedIndex() {
                 return Collections.unmodifiableList(loadedIndex);
             }
+
+            @Override
+            public <T> void onProvision(ProvisionInvocation<T> provision) {
+                final T instance = provision.provision();
+                if (instance == null) return;
+                List<Method> methods = new LinkedList<>();
```

Comment:
Sorting order is not clear here. It should be somehow documented so that the approach remains compatible if algorithm changes. Maybe `ordinals` will help, and `load()` must have high ordinal then